homeassistant/components/konnected/config_flow.py

```diff
@@ -302,114 +255,65 @@ class KonnectedFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+        This will raise CannotConnect if an error occurs
+        """
+        self.data[CONF_HOST] = host
+        self.data[CONF_PORT] = port
+        try:
+            status = await get_status(self.hass, host, port)
+            self.data[CONF_ID] = status["mac"].replace(":", "")
+            self.data[CONF_MODEL] = status.get("name", KONN_MODEL)
+            self.data[CONF_ACCESS_TOKEN] = "".join(
+                random.choices(f"{string.ascii_uppercase}{string.digits}", k=20)
+            )
 
-    @callback
-    def async_cache_config(self, config):
-        """Cache the device config for shared usage."""
-        domain_data = self.hass.data.setdefault(DOMAIN, {"config_data": {}})
-        domain_data["config_data"][self.device_id] = config
+        except (CannotConnect, KeyError):
+            raise CannotConnect
 
-    async def async_step_import(self, import_info):
+    async def async_step_import(self, device_config):
         """Import a configuration.yaml config.
 
         This flow is triggered by `async_setup` for configured panels.
-        Triggered for any panel that does not have a config entry
-        yet (based on device_id).  If the cfg entry can't be made
-        we'll cache the data in hass.data[DOMAIN]["config_data"] for
-        other flows to reference.
         """
-        _LOGGER.debug(import_info)
-        try:
-            device_config = DEVICE_SCHEMA_YAML(import_info)
-
-        except vol.Invalid as err:
-            _LOGGER.error(
-                "Cannot import config..%s", humanize_error(import_info, err),
-            )
-            return self.async_abort(reason="unknown")
-
-        # swap out pin for zones in a io config
-        def pins_to_zones(config):
-            for zone in config:
-                if zone.get(CONF_PIN):
-                    zone[CONF_ZONE] = PIN_TO_ZONE[zone[CONF_PIN]]
-                    del zone[CONF_PIN]
+        _LOGGER.debug(device_config)
 
+        # populate any options
+        device_config[CONF_IO] = {}
         if device_config.get(CONF_BINARY_SENSORS):
             pins_to_zones(device_config[CONF_BINARY_SENSORS])
+            for io_cfg in device_config[CONF_BINARY_SENSORS]:
+                device_config[CONF_IO][io_cfg[CONF_ZONE]] = CONF_IO_BIN
 
         if device_config.get(CONF_SENSORS):
             pins_to_zones(device_config[CONF_SENSORS])
+            for io_cfg in device_config[CONF_SENSORS]:
+                device_config[CONF_IO][io_cfg[CONF_ZONE]] = CONF_IO_DIG
 
         if device_config.get(CONF_SWITCHES):
             pins_to_zones(device_config[CONF_SWITCHES])
+            for io_cfg in device_config[CONF_SWITCHES]:
+                device_config[CONF_IO][io_cfg[CONF_ZONE]] = CONF_IO_SWI
+
+        # save the data and confirm connection via user step
+        await self.async_set_unique_id(device_config["id"])
+        self.options = OPTIONS_SCHEMA(device_config)
+        if device_config.get(CONF_HOST) and device_config.get(CONF_PORT):
```

Comment:
Port doesn't seem part of the config schema. We should mark host and port inclusive with `vol.Inlusive` in the config schema if both are required if any is present.

Reply:
Good catch - will do

Reply:
Can we simplify this check now, after we added `vol.Inclusive` to the config schema?

Reply:
Yea - seems reasonable to just check for one.  I like the code clarity of checking both but a comment will make up for it.